src/tracing/core/shared_memory_arbiter_impl.cc

```diff
@@ -908,6 +908,64 @@ void SharedMemoryArbiterImpl::NotifyFlushComplete(FlushRequestID req_id) {
   }
 }
 
+void SharedMemoryArbiterImpl::ScrapeEmulatedSharedMemoryBuffer(
+    const std::map<WriterID, BufferID>& buffer_for_writers) {
+  if (!use_shmem_emulation_)
```

Comment:
I think this should be a PERFETTO_CHECK(). we should never get here without shmem emulation, if we do it's  a bug and we should fail fast, not hide it